src/rpcnet.cpp

```diff
@@ -443,3 +443,79 @@ Value getnetworkinfo(const Array& params, bool fHelp)
     obj.push_back(Pair("localaddresses", localAddresses));
     return obj;
 }
+
+Value setban(const Array& params, bool fHelp)
+{
+    string strCommand;
+    if (params.size() >= 2)
+        strCommand = params[1].get_str();
+    if (fHelp || params.size() < 2 ||
+        (strCommand != "add" && strCommand != "remove"))
+        throw runtime_error(
+                            "setban \"node\" \"add|remove\" (bantime)\n"
+                            "\nAttempts add or remove a IP from the banned list.\n"
+                            "\nArguments:\n"
+                            "1. \"ip\"       (string, required) The IP (see getpeerinfo for nodes ip)\n"
+                            "2. \"command\"  (string, required) 'add' to add a IP to the list, 'remove' to remove a IP from the list\n"
+                            "1. \"bantime\"  (numeric, optional) time in seconds how long the ip is banned\n"
+                            "\nExamples:\n"
+                            + HelpExampleCli("setban", "\"192.168.0.6\" \"add\" 86400")
+                            + HelpExampleRpc("setban", "\"192.168.0.6\", \"add\" 86400")
+                            );
+
+    string strNode = params[0].get_str();
+    CNetAddr netAddr(strNode);
+    if (!netAddr.IsValid())
+        throw JSONRPCError(RPC_CLIENT_NODE_ALREADY_ADDED, "Error: Invalid IP Address");
+
+    if (strCommand == "add")
+    {
+        if (CNode::IsBanned(netAddr))
+            throw JSONRPCError(RPC_CLIENT_NODE_ALREADY_ADDED, "Error: IP already banned");
+
+        int64_t banTime = 0; //use standard bantime if not specified
+        if (params.size() == 3 && !params[2].is_null())
+            banTime = params[2].get_int64();
+
+        CNode::Ban(netAddr, banTime);
+
+        //disconnect possible nodes
+        while(CNode *bannedNode = FindNode(netAddr))
+            bannedNode->CloseSocketDisconnect();
+    }
+    else if(strCommand == "remove")
```

Comment:
Agreed on the remove all feature. But it would be ugly to have a command like `setban <IP> removeall` (<IP> would then be dummy). And supporting `setban removeall` will make the help message unreadable (because of parameter mixups).
IMO adding a `clearbanned` command could make sense.
Any other suggestions?